docs/user-guide/configure.md

```diff
@@ -2,14 +2,46 @@
 
 Stylelint expects a configuration object, and looks for one in a:
 
+- `stylelint.config.js` or `.stylelintrc.js` file
+  - Which module system to use depends on your [default module system configuration](https://nodejs.org/api/packages.html#determining-module-system) for Node.js (e.g., `"type": "module"` in `package.json`).
+- `stylelint.config.mjs` or `.stylelintrc.mjs` file using `export default` (ES module)
+- `stylelint.config.cjs` or `.stylelintrc.cjs` file using `module.exports` (CommonJS)
+- `.stylelintrc.json`, `.stylelintrc.yml`, or `.stylelintrc.yaml` file
+- `.stylelintrc` file in JSON or YAML format
+  - We recommend adding an extension (e.g., `.json`) to help your editor provide syntax checking and highlighting.
 - `stylelint` property in `package.json`
-- `.stylelintrc` file
-- `.stylelintrc.{cjs,js,json,yaml,yml}` file
-- `stylelint.config.{cjs,mjs,js}` file
```

Comment:
[note]

- Move `stylelint.config.js` to the top because we are considering a flat configuration in a future major release
- Move `stylelint` property (in `package.json`) to the bottom because it's likely to be less used